orca-interlink/src/main/java/com/netflix/spinnaker/orca/interlink/MessageFlagger.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.interlink
+
+import com.google.common.base.Preconditions
+import com.google.common.collect.EvictingQueue
+import com.google.common.hash.HashCode
+import com.google.common.hash.Hashing
+import com.netflix.spinnaker.orca.interlink.events.InterlinkEvent
+
+class MessageFlagger(maxSize: Int, val threshold: Int) {
+  val queue: EvictingQueue<HashCode> = EvictingQueue.create(maxSize)
+  val hasher = Hashing.goodFastHash(64)
+
+  init {
+    Preconditions.checkArgument(maxSize >= threshold, "maxSize (%s) has to be larger than threshold (%s)", maxSize, threshold)
+    Preconditions.checkArgument(threshold > 0, "threshold (%s) has to be > 0", threshold)
+  }
+
+  fun isFlagged(event: InterlinkEvent): Boolean {
+    val hashCode = hasher.hashString("${event.eventType}:${event.executionId}:${event.executionType}", Charsets.UTF_8)
```

Comment:
i still think we need the "original submission" timestamp here.

Also, do you even need the hashcode here? you can just push the fingerprint into the queue - I don't think the hash buys you much, these are pretty short strings

Reply:
The strings are pretty short for now, but with the hashcode I feel it's a little more generic. Each fingerprint is also going to be hashed only once but compared dozens of times, so I don't find it shocking

Will add some kind of timestamp based on our offline discussion 👍

Reply:
made it more clear that I intend the fingerprint to be more opaque/generic/subject to change here: 6e4393260bba1435e7c19ea2c3b911cfc4e2a10c